test/functional/rpc_misc.py

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env python3
+# Copyright (c) 2018 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""Test RPC misc output."""
+import xml.etree.ElementTree as ET
+
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import (
+    assert_raises_rpc_error,
+    assert_equal
+)
+
+from test_framework.authproxy import JSONRPCException
+
+
+class RpcMiscTest(BitcoinTestFramework):
+    def set_test_params(self):
+        self.num_nodes = 1
+
+    def run_test(self):
+        node0 = self.nodes[0]
+
+        self.log.info("test getmemoryinfo")
+        memory = node0.getmemoryinfo()['locked']
+        assert memory['used'], memory['free'] > 0
+        assert memory['total'] > 0
+        assert memory['locked'] > 0
+        assert memory['chunks_used'] > 0
+        assert memory['chunks_free'] > 0
+        assert_equal(memory['used'] + memory['free'], memory['total'])
+
+        self.log.info("test mallocinfo")
+        try:
+            mallocinfo = node0.getmemoryinfo(mode="mallocinfo")
+            self.log.info('getmemoryinfo(mode="mallocinfo") call succeeded')
+            tree = ET.fromstring(mallocinfo)
+            assert_equal(tree.tag, 'malloc')
+        except JSONRPCException:
+            self.log.info('getmemoryinfo(mode="mallocinfo") not available')
+            assert_raises_rpc_error(-8, 'mallocinfo is only available when compiled with glibc 2.10+', node0.getmemoryinfo, mode="mallocinfo")
+
+        assert_raises_rpc_error(-8, "unknown mode " + "foobar", node0.getmemoryinfo, mode="foobar")
```

Comment:
nit: no need to concat two string literals. Do `"unknown mode foobar"` instead of `"unknown mode " + "foobar"`